sql/catalyst/src/main/scala/org/apache/spark/sql/types/DecimalType.scala

```diff
@@ -136,10 +137,54 @@ object DecimalType extends AbstractDataType {
     case DoubleType => DoubleDecimal
   }
 
+  private[sql] def forLiteral(literal: Literal): DecimalType = literal.value match {
```

Comment:
Is this different than `forType` if applied on `Literal.dataType`?

Reply:
yes, please see my comment above for an example. Thanks.